In [ ]:
#Data Preprocessing
import pandas as pd
import numpy as np

DGS2 = pd.read_csv('data/DGS2.csv')
DGS5 = pd.read_csv('data/DGS5.csv')
DGS10 = pd.read_csv('data/DGS10.csv')    

print(DGS2.shape, DGS5.shape, DGS10.shape) #Print the shapes of the dataframes to verify successful loading
print(DGS2.info(),DGS5.info(), DGS10.info()) #Print the info of the dataframe to check data types and non-null counts

merged = DGS2.merge(DGS5, on="observation_date", how="inner").merge(DGS10, on="observation_date", how="inner") #Merge the dataframes on observation_date
merged = merged.dropna(subset=["DGS2","DGS5","DGS10"]) #Drop rows with missing values in any of the yield columns

merged = merged.set_index("observation_date") #Set observation_date as index
merged.index = pd.to_datetime(merged.index, errors="coerce") #Convert index to datetime



(6996, 2) (6996, 2) (6996, 2)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6996 entries, 0 to 6995
Data columns (total 2 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   observation_date  6996 non-null   object 
 1   DGS2              6709 non-null   float64
dtypes: float64(1), object(1)
memory usage: 109.4+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6996 entries, 0 to 6995
Data columns (total 2 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   observation_date  6996 non-null   object 
 1   DGS5              6709 non-null   float64
dtypes: float64(1), object(1)
memory usage: 109.4+ KB
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6996 entries, 0 to 6995
Data columns (total 2 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   observation_date  6996 non-null   object 
 1   DGS10             6709

# Preprocessing summary

-Loaded raw FRED CSVs (DGS2.csv, DGS5.csv, DGS10.csv) and inspected shapes/dtypes. We saw 6996 entries of which 6709 were non-null

-Merged the three series on the common key observation_date using an inner join.

-Dropped rows with missing yields in any of DGS2, DGS5, DGS10 (missing days are typically U.S. market holidays or occasional gaps) leaving us with the 6709 entries from before

-Converted observation_date to proper datetime and set it as the index to enable clean time-series operations

,DGS2,DGS5,DGS10
observation_date,,,
1999-01-04,4.58,4.57,4.69
1999-01-05,4.64,4.62,4.74
1999-01-06,4.62,4.61,4.73
1999-01-07,4.63,4.62,4.77
1999-01-08,4.71,4.72,4.86
...,...,...,...
2025-10-21,3.45,3.56,3.98
2025-10-22,3.45,3.56,3.97
2025-10-23,3.48,3.61,4.01


In [28]:
merged.info()

<class 'pandas.core.frame.DataFrame'>
DatetimeIndex: 6709 entries, 1999-01-04 to 2025-10-27
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   DGS2    6709 non-null   float64
 1   DGS5    6709 non-null   float64
 2   DGS10   6709 non-null   float64
dtypes: float64(3)
memory usage: 209.7 KB
